### ISSUE TITLE
Difference between HAA and HAA OTA releases?

### ISSUE THREAD

**Issue by wrx5** [RavenSystem/esp-homekit-devices#1243]
Hi Everyone,

Forgive me for this "dummies" question but I wonder What's the difference between HAA and HAA OTA releases?

Today the latest version of HAA is 3.8.4 and HAA OTA is 4.3.0. 

When I put the device into setup mode and tick the Update All Now and Auto OTA Updates options then hit "save" button it only gets HAA releases not HAA OTA.

Thanks in advance

**Comment by rbrtsluk** [comment]
no one knows?

**Comment by tonysprenk** [comment]
HAA OTA is the part of the software that manages the update and install mechanism. There always needs to be two parts (HAA OTA and HAA) because when you update, HAA is overwritten with the new version. But if there is no HAA OTA there is no software to tell HAA where and how to be installed anymore. This is the simplest way I can think of to explain it, hopefully it makes sense.

**Comment by wrx5** [comment]
So where do we use HAA OTA? can we flash the device with that one instead of HAA?

**Comment by tonysprenk** [comment]
It’s part of fullhaaboot.bin so if you flash with that it’s included. The other option is to flash without OTA (only in special cases). More info here: https://github.com/RavenSystem/esp-homekit-devices/wiki/Installation

**Comment by xrust83** [comment]
https://www.samsung.com/my/support/mobile-devices/what-are-the-requirements-when-upgrade-firmware-with-fota-firmware-over-the-air/

**Comment by ss83** [comment]
I used the HomeKitFlasher for two magichome devices. The first one I used the "Flash HAA v.4.XXX" button. The second one I used the "Flash OTA HAA Firmware".

Am I right that only the second one where I used the OTA button will be able to update through "AUTO OTA Updates" ? Is there a way to update the first one to the OTA version using the interface, e.g. under the custom server ??? I have already installed it into the wall, so it would be nice not to dismantle it again :-)
